accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -1191,10 +1226,17 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
                                 }
                             }
                         }
+
+                        Some(*key)
+                    } else {
+                        // Entry was not dirty at scan time and had ref_count == 1
+                        // Skip all checks (including should_evict_from_mem) and do not do any disk ops
+                        // Pass directly to evict_from_cache, which will re-check conditions under write lock
+                        Some(*key)
```

Comment:
Since this is so short, can this (clean) be the `if` case, and then 'dirty' can be the else?

Reply:
done.